src/net_processing.cpp

```diff
@@ -223,6 +223,9 @@ struct Peer {
     /** Services this peer offered to us. */
     std::atomic<ServiceFlags> m_their_services{NODE_NONE};
 
+    //! Whether this peer is an inbound connection
+    const bool m_is_inbound;
```

Comment:
The primary source of that information is `CNode::m_conn_type`. Is it necessary to have a partial copy of that in `Peer`?
If not, an alternative approach might be to use that, as is done for various other spots in net_processing that call `IsInboundConn()`. We might need also need helper functions similar to `GetExtraBlockRelayCount` but for inbounds. I haven't tried it or looked into it in depth, but just wanted to bring up where you've considered that option.

Reply:
We do, at least to my understanding, given some `PeerMan` methods that are not called with any `CNode` reference need to check whether the peer that sent us the data they are processing is inbound or outbound.

A good example is `PeerManagerImpl::MaybeSetPeerAsAnnouncingHeaderAndIDs`, which is called from `PeerManagerImpl::BlockChecked` and is passed a `NodeId` grabbed from `mapBlockSource`

Reply:
Yes, but we could attempt to retrieve the `CNode` by `NodeId` from `m_nodes`, as the current code does it for `Peer` from `m_peer_map`.
I think the lookup would be `O(N)` for iterating through the vector `m_nodes` instead of `O(log N)` for the `std::map`, so maybe that is the reason for the redundancy?